src/pages/signin/LoginForm.js

```diff
@@ -65,165 +65,143 @@ const defaultProps = {
-                >
-                    <TextInput
-                        ref={(el) => (this.input = el)}
-                        label={this.props.translate('loginForm.phoneOrEmail')}
-                        value={this.state.login}
-                        autoCompleteType="username"
-                        textContentType="username"
-                        nativeID="username"
-                        name="username"
-                        onChangeText={this.onTextInput}
-                        onSubmitEditing={this.validateAndSubmitForm}
-                        autoCapitalize="none"
-                        autoCorrect={false}
-                        keyboardType={CONST.KEYBOARD_TYPE.EMAIL_ADDRESS}
-                        errorText={formErrorText}
-                        hasError={hasError}
-                    />
-                </View>
-                {!_.isEmpty(this.props.account.success) && <Text style={[styles.formSuccess]}>{this.props.account.success}</Text>}
-                {!_.isEmpty(this.props.closeAccount.success || this.props.account.message) && (
-                    // DotIndicatorMessage mostly expects onyxData errors, so we need to mock an object so that the messages looks similar to prop.account.errors
-                    <DotIndicatorMessage
-                        style={[styles.mv2]}
-                        type="success"
-                        messages={{0: this.props.closeAccount.success || this.props.account.message}}
-                    />
-                )}
-                {
-                    // We need to unmount the submit button when the component is not visible so that the Enter button
-                    // key handler gets unsubscribed and does not conflict with the Password Form
-                    this.props.isVisible && (
-                        <View style={[styles.mt5]}>
-                            <FormAlertWithSubmitButton
-                                buttonText={this.props.translate('common.continue')}
-                                isLoading={this.props.account.isLoading && this.props.account.loadingForm === CONST.FORMS.LOGIN_FORM}
-                                onSubmit={this.validateAndSubmitForm}
-                                message={serverErrorText}
-                                isAlertVisible={!_.isEmpty(serverErrorText)}
-                                containerStyles={[styles.mh0]}
-                            />
-                        </View>
-                    )
-                }
-            </>
-        );
-    }
+        Session.beginSignIn(parsedPhoneNumber.possible ? parsedPhoneNumber.number.e164 : loginTrim);
+    }, [login, closeAccount, network, setFormError]);
+
+    useEffect(() => {
+        Session.clearAccountMessages();
+    }, []);
+
+    useEffect(() => {
+        if (!canFocusInputOnScreenFocus() || !input.current || !isVisible) {
+            return;
+        }
+        if (blurOnSubmit) {
```

Comment:
combine both did mount and did update - may potentially trigger input.current.blur(); in situations where it should not, for example anything changed in input ref, which will trigger use effect and if blurOnSubmit will be exists - user will see blur and after focus event on inputs, which is not correct. I will suggest to separate this 1 useEffect to 2

Reply:
Hi, @narefyev91 
Separated 1 useEffect to 2 useEffect following your suggestion.
Please check.